be/src/exec/pipeline/fragment_context.cpp

```diff
@@ -138,6 +138,10 @@ void FragmentContext::set_stream_load_contexts(const std::vector<StreamLoadConte
 }
 
 void FragmentContext::cancel(const Status& status) {
+    if (_runtime_state != nullptr && _runtime_state->query_ctx() != nullptr) {
+        _runtime_state->query_ctx()->release_workgroup_token_once();
```

Comment:
Because we only have `cancel_fragment_instance` RPC, there isn't `cancel_query` RPC interface.
And `release_workgroup_token_once` guarantee that it can be only released once.

Reply:
And here don't cancel query context, it just release workgroup token, which is owned by the query context.